File: backend/FleetMind.Api/Models/Port.cs
using System;
using FleetMind.Api.Models.Common;

namespace FleetMind.Api.Models
{
    public class Port : BaseEntity
    {
        public string Name { get; set; } = string.Empty;
        public string UnLocode { get; set; } = string.Empty;
        public string Country { get; set; } = string.Empty;
        public string City { get; set; } = string.Empty;
        public decimal? Latitude { get; set; }
        public decimal? Longitude { get; set; }
    }
}

File: frontend/src/types/port.ts
import { PaginationQuery } from './pagination';

export interface Port {
  id: string;
  name: string;
  unLocode: string;
  country: string;
  city: string;
  latitude: number | null;
  longitude: number | null;
  createdAt: string;
}

export interface CreatePortPayload {
  name: string;
  unLocode: string;
  country: string;
  city: string;
  latitude?: number;
  longitude?: number;
}

export interface UpdatePortPayload {
  name: string;
  country: string;
  city: string;
  latitude?: number;
  longitude?: number;
}

export interface PortQuery extends PaginationQuery {
  country?: string;
}
